platform/windows/os_windows.cpp

```diff
@@ -101,7 +101,7 @@ static String fix_path(const String &p_path) {
 	}
 	path = path.simplify_path();
 	path = path.replace("/", "\\");
-	if (!path.is_network_share_path() && !path.begins_with(R"(\\?\)")) {
+	if (!path.is_network_share_path() && !path.begins_with(R"(\\?\)") && path.size() >= MAX_PATH) {
```

Comment:
Doesn't matter much, but I feel like the new check could be the first, and shortcircuit the rest if this necessary condition isn't met.
```suggestion
	if (path.size() >= MAX_PATH && !path.is_network_share_path() && !path.begins_with(R"(\\?\)")) {
```